Orange/widgets/visualize/tests/test_owsieve.py

```diff
@@ -59,3 +59,22 @@ def test_chisquare(self):
         table = Table(Domain([a, b]), list(zip("yynny", "ynyyn")))
         chi = ChiSqStats(table, 0, 1)
         self.assertFalse(isnan(chi.chisq))
+
+    def test_metadata(self):
+        """
+        Widget should intepret meta data which are continuous or discrete in the same way
+        as features or target. However still one variable should be target or feature.
+        gh-2098
+        """
+        table = Table(
+            Domain(
+                [],
+                [],
+                [ContinuousVariable("a"),
+                 DiscreteVariable("b", values=["y", "n"])]
+            ),
+            list(zip(
+                [42.48, 16.84, 15.23, 23.8],
+                "yynn"))
+        )
+        self.send_signal("Data", table)
```

Comment:
This checks just that the widget does not crash. It is difficult to test this widget assuming anything about the widget internals. For the lack of better options, I'd replace `self.send_signal("Data", table)` with

```
        from unittest.mock import patch
        from Orange.widgets.visualize.owsieve import Discretize
        with patch("Orange.widgets.visualize.owsieve.Discretize",
                   wraps=Discretize) as disc:
            self.send_signal("Data", table)
            self.assertTrue(disc.called)
        metas = self.widget.discrete_data.domain.metas
        self.assertEqual(len(metas), 2)
        self.assertTrue(all(attr.is_discrete for attr in metas))
```

Move imports to the beginning of the file, of course.